File: 7_java/EsPalindromo2.java
public class EsPalindromo2 {
  public static boolean esPalindromo2(String palabra) {	
    int izquierda = 0;
    int derecha = palabra.length() - 1;
    while (izquierda < derecha) {
      if (palabra.toLowerCase().charAt(izquierda) != palabra.toLowerCase().charAt(derecha)) {
	return false;
      }
      izquierda++;
      derecha--;
    }
    return true;
  }

  public static void main(String[] args) {
    System.out.println(esPalindromo2("mex"));
    System.out.println(esPalindromo2("Lol"));
  }
}

File: 7_typescript/esPalindromo2.ts
export function esPalindromo2(palabra: string) {
  let izquierda = 0;
  let derecha = palabra.length - 1;
  while (izquierda < derecha) {
    if (palabra[izquierda].toLowerCase() !== palabra[derecha].toLowerCase())
      return false;
    izquierda++;
    derecha--;
  }
  return true;
}

console.log(esPalindromo2("mex"));
console.log(esPalindromo2("Lol"));
